app/facades/restaurant_facade.rb

```diff
@@ -0,0 +1,17 @@
+class RestaurantFacade
+  def initialize(activity, latitude, longitude)
+    @activity = activity
+    @lat = latitude
+    @long = longitude
+  end
```

Comment:
You can actually not have a search term BUT restrict the category to be only restaurants. So here `activity` could be default to nil and alter your service to take categories